scripts/FCCee/ALLEGRO/ALLEGRO_o1_v03/ALLEGRO_make_TH1.py

```diff
@@ -0,0 +1,154 @@
+import ROOT
+from podio import root_io
+import dd4hep as dd4hepModule
+from ROOT import dd4hep
+
+import argparse
+
+import os
+from dd4hep import Detector
+import DDRec
+
+
+def make_TH1_file(args):
+
+  # create output ROOT file
+  outputFile = ROOT.TFile(args.outputFile, "RECREATE")
+
+  # set file reader
+  podio_reader = root_io.Reader(args.inputFile)
+
+  # set list of directories in ROOT file (one per subsystem to check)
+  dir_list = []
+  # set list of (list of) histograms created (one list per subsystem to check)
+  histo_list = []
+
+  # get detector description for cell id decoding
+  #theDetector = Detector.getInstance()
+  #theDetector.fromXML(os.environ["$K4GEO"]+"/FCCee/ALLEGRO/compact/ALLEGRO_o1_v03/ALLEGRO_o1_v03.xml") 
+  #idposConv = DDRec.CellIDPositionConverter(theDetector)
+                  
+
+
+
+  #############################################################################
+  ##                                                                         ## 
+  ##                BEGIN: ECal Barrel Histogram Definition                  ##
+  ##                                                                         ## 
+  #############################################################################
+
```

Comment:
I'm not a fan of these blocks, if you keep reading then it's obvious so it's not telling you anything. If you want a small comment so that you can jump, it can be just `# Beginning of ECal Barrel` and the ending has `End`so that the in the editor you can search for `Begin` and `End`. In one of the files below, the blocks look to be almost 50% of the contents of the script.

Reply:
It might also be a simple sign that you effectively have (almost) a function (or at least something that can be refactored into one) here. In that case you solve several issues in one step:
- You have a clearly defined name for searching
- You have a very well defined scope of where it starts and ends
- You have a clearly defined place where you can put some top level comments / a docstring
- Most editors allow you to fold / collapse the function body, so readability of the complete file also improves

Reply:
I put them because Brieuc requested them: we wanted clearly separated sections for each subdetector so people could easily add new sections or modify the ones they were responsible for without disturbing the rest of the script. I can try to make them smaller if you think that would help, but i would not get rid of them completely.

Encapsulating them into functions is also an option, let me know what would be preferred.

Reply:
I would have a preference for functions, as that minimizes chances that changes in one "section" influence anything somewhere else.